Source/cursor.cpp

```diff
@@ -366,7 +366,7 @@ void CheckCursMove()
 	if ((sgbMouseDown != CLICK_NONE || ControllerActionHeld != GameActionType_NONE) && IsNoneOf(LastMouseButtonAction, MouseActionType::None, MouseActionType::Attack, MouseActionType::Spell)) {
 		InvalidateTargets();
 
-		if (pcursmonst == -1 && ObjectUnderCursor == nullptr && pcursitem == -1 && pcursinvitem == -1 && pcursstashitem == uint16_t(-1) && pcursplr == -1) {
+		if (pcursmonst == -1 && ObjectUnderCursor == nullptr && pcursitem == -1 && pcursinvitem == -1 && pcursstashitem == -1 && pcursplr == -1) {

```

Comment:
This line is likely failing the timedemo.

Assigning -1 to a uint16_t (unsigned int) is not the same as comparing a unsigned int 16 with a signed 32 bit int.

Run the following example:
```
uint16_t test = -1;
if (test == -1)
	Log("What not the same?!");
```
`What not the same?!` will be printed. 😉 
uin16_t will be changed/widened to `int32_t` for comparison.
But assigning will use `uint16_t` casted -1.

That's why
> Uniform invalid item id usage: This is an opinionated change to clarify the code at a glance. While it is functionally the same as uint16_t(-1), it feels like a more obvious explanation of what is happening.

is not correct.

But your statement that `uint16_t(-1)` is not obvious is still true.
My suggestion: instead of `uint16_t(-1)` we could use `StashStruct::EmptyCell`. This would make the code more readable and still works fine. 🙂

Reply:
Reverted to `-1` for now as I am not found of EmptyCell being used sporadically. I feel like a globally defined "invalid item id" should exist, but this PR has been going too long and needs to be completed without anymore scope creep. 

That aside, thank you for the excellent explanation!